src/lib/mina_base/test/zkapp_command/call_forest.ml

```diff
@@ -0,0 +1,320 @@
+open Core_kernel
+open Signature_lib
+open Mina_base
+open Mina_base.Zkapp_command.Call_forest
+
+(* TODO: move generators to the library once the code isn't actively being
+     worked on *)
+module Tree = struct
+  include Tree
+
+  module Stable = struct
+    include Stable
+
+    module V1 = struct
+      include V1
+
+      let quickcheck_generator account_update_gen account_update_digest_gen
+          digest_gen =
+        let open Quickcheck.Generator.Let_syntax in
+        Quickcheck.Generator.fixed_point (fun self ->
+            let%bind calls_length =
+              Quickcheck.Generator.small_non_negative_int
+            in
+            let%map account_update = account_update_gen
+            and account_update_digest = account_update_digest_gen
+            and digest = digest_gen
+            and calls =
+              Quickcheck.Generator.list_with_length calls_length
+                (With_stack_hash.quickcheck_generator self digest_gen)
+            in
+            { account_update; account_update_digest; calls } )
+    end
+  end
+end
+
+module Shape = struct
+  include Shape
+
+  let rec sexp_of_t = function
```

Comment:
Instead of including `Shape`, and writing `sexp_of_t`, it would be more concise to have:
```
  type t = Shape.t = Node of (Shape.I.t * t) list [@@deriving sexp_of]
```
and also add the deriver to `Shape.I.t`. Less code, easier to maintain.

Reply:
Yes, that's what is going to be done once these changes are moved into the original module. The type is not redefined here (and therefore, the derivers aren't being used) because redefining the type would cause type errors later.

Reply:
In what I wrote, there's the constraint that it's the same type as `Shape.t`, so there shouldn't be any type errors.